### PR TITLE
Add tests for closed-realm access protection

### PR DESCRIPTION
**Pull request by kristiandupont**

### CODE REVIEW

**Comment by alazier** [comment]
We need to test all methods/properties available on realm including transactions.

File: src/js_realm.hpp [hunk 1]
```diff
@@ -639,6 +639,9 @@ void RealmClass<T>::add_listener(ContextType ctx, ObjectType this_object, size_t
     auto callback = Value::validated_to_function(ctx, arguments[1]);
 
     SharedRealm realm = *get_internal<T, RealmClass<T>>(this_object);
+    if (realm->is_closed()) {
```
Comment (alazier) [inline src/js_realm.hpp]: should we check for remove listener and remove all as well?

Comment (kristiandupont) [inline src/js_realm.hpp]: Yeah, let me add that.

**Comment by kristiandupont** [comment]
Yes, we should probably merge the https://github.com/realm/realm-object-store/pull/278 PR into our JS branch

**Pull request merged**